packages/@lwc/engine/src/framework/observed-fields.ts

```diff
@@ -27,9 +27,8 @@ export function createObservedFieldPropertyDescriptor(key: string): PropertyDesc
 
             if (newValue !== vm.cmpFields[key]) {
                 vm.cmpFields[key] = newValue;
-                if (isFalse(vm.isDirty)) {
-                    valueMutated(this, key);
-                }
+
+                valueMutated(vm, key);
```

Comment:
this is changing the semantic of the method name... maybe the local mutation-tracker file should expose something called `componentValueMutated(vm, key)` while the original continue to be `valuedMutated()`